lib/components/Geojson.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import MapView from './MapView';
+import uuid from 'uuid';
+
+export const makeOverlays = features => {
+    const points = features
+        .filter(f => f.geometry && (f.geometry.type === 'Point' || f.geometry.type === 'MultiPoint'))
+        .map(feature => makeCoordinates(feature).map(coordinates => makeOverlay(coordinates, feature)))
+        .reduce(flatten, [])
+        .map(overlay => ({ ...overlay, type: 'point' }));
+
+    const lines = features
+        .filter(
+            f => f.geometry && (f.geometry.type === 'LineString' || f.geometry.type === 'MultiLineString')
+        )
+        .map(feature => makeCoordinates(feature).map(coordinates => makeOverlay(coordinates, feature)))
+        .reduce(flatten, [])
+        .map(overlay => ({ ...overlay, type: 'polyline' }));
+
+    const multipolygons = features
+        .filter(f => f.geometry && f.geometry.type === 'MultiPolygon')
+        .map(feature => makeCoordinates(feature).map(coordinates => makeOverlay(coordinates, feature)))
+        .reduce(flatten, []);
+
+    const polygons = features
+        .filter(f => f.geometry && f.geometry.type === 'Polygon')
+        .map(feature => makeOverlay(makeCoordinates(feature), feature))
+        .reduce(flatten, [])
+        .concat(multipolygons)
+        .map(overlay => ({ ...overlay, type: 'polygon' }));
+
+    return points.concat(lines).concat(polygons);
+};
+
+const flatten = (prev, curr) => prev.concat(curr);
+
+const makeOverlay = (coordinates, feature) => {
+    let overlay = {
+        feature,
+        id: feature.id ? feature.id : uuid(),
```

Comment:
Does this use of uuid really make sense? It will be a new uuid on every pass. Same further down.

Reply:
Id is used for key prop of array of React components.

Reply:
Right, `uuid` doesn't make any sense here. You could use `index` in
```
overlays.map((overlay, index) => {
  key={overlay.id || index}
}
```
Also it isn't necessary in
```
<React.Fragment key={props.geojson.id || uuid()}>
```

Reply:
uuid is a quite costly operation, it shouldn't be used everywhere